src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -51,711 +78,279 @@ public class AnalysisService : IDisposable
+        public void OnConfigurationUpdated(object sender, LanguageServerSettings settings)
         {
-            // Save ourselves some work here
-            if (!_logger.IsEnabled(LogLevel.Debug))
-            {
-                return;
-            }
-
-            // If we already know the module that was imported, save some work
-            if (_pssaModuleInfo == null)
-            {
-                PowerShellResult getModuleResult = InvokePowerShell(
-                    "Get-Module",
-                    new Dictionary<string, object>{ {"Name", PSSA_MODULE_NAME} });
-
-                if (getModuleResult == null)
-                {
-                    throw new AnalysisServiceLoadException("Get-Module call to find PSScriptAnalyzer module failed");
-                }
-
-                _pssaModuleInfo = getModuleResult.Output
-                    .Select(m => m.BaseObject)
-                    .OfType<PSModuleInfo>()
-                    .FirstOrDefault();
-            }
-
-            if (_pssaModuleInfo == null)
-            {
-                throw new AnalysisServiceLoadException("Unable to find loaded PSScriptAnalyzer module for logging");
-            }
-
-            var sb = new StringBuilder();
-            sb.AppendLine("PSScriptAnalyzer successfully imported:");
-
-            // Log version
-            sb.Append("    Version: ");
-            sb.AppendLine(_pssaModuleInfo.Version.ToString());
-
-            // Log exported cmdlets
-            sb.AppendLine("    Exported Cmdlets:");
-            foreach (string cmdletName in _pssaModuleInfo.ExportedCmdlets.Keys.OrderBy(name => name))
-            {
-                sb.Append("    ");
-                sb.AppendLine(cmdletName);
-            }
-
-            // Log available rules
-            sb.AppendLine("    Available Rules:");
-            foreach (string ruleName in GetPSScriptAnalyzerRules())
-            {
-                sb.Append("        ");
-                sb.AppendLine(ruleName);
-            }
-
-            _logger.LogDebug(sb.ToString());
+            ClearOpenFileMarkers();
+            _analysisEngine = InstantiateAnalysisEngine();
         }
 
-        private async Task<PSObject[]> GetDiagnosticRecordsAsync<TSettings>(
-             string scriptContent,
-             string[] rules,
-             TSettings settings) where TSettings : class
+        private PssaCmdletAnalysisEngine InstantiateAnalysisEngine()
         {
-            // When a new, empty file is created there are by definition no issues.
-            // Furthermore, if you call Invoke-ScriptAnalyzer with an empty ScriptDefinition
-            // it will generate a ParameterBindingValidationException.
-            if (string.IsNullOrEmpty(scriptContent)
-                || !(typeof(TSettings) == typeof(string) || typeof(TSettings) == typeof(Hashtable)))
+            if (!(_configurationService.CurrentSettings.ScriptAnalysis.Enable ?? false))
```

Comment:
I'm curious. If you do `.Enable.Value` here, will you get an exception or will it use the default value for bool. If the latter, I'd prefer the `.Value` instead of the `?? false`

Reply:
[It throws](https://github.com/dotnet/runtime/blob/92025b2c5a23ce317fb5ff5432213e465d3f84f1/src/libraries/System.Private.CoreLib/src/System/Nullable.cs#L35-L45)